yew/src/services/websocket.rs

```diff
@@ -172,13 +180,21 @@ impl WebSocketService {
         &mut self,
         url: &str,
         notification: &Callback<WebSocketStatus>,
-    ) -> Result<ConnectCommon, &str> {
+    ) -> Result<ConnectCommon, WebSocketError> {
         let ws = WebSocket::new(url);
-        if ws.is_err() {
-            return Err("Failed to created websocket with given URL");
-        }
+        #[cfg(feature="web_sys")]
+        if let Err(ws_error) = &ws {
+            return Err(WebSocketError::CreationError(ws_error.clone().unchecked_into::<js_sys::Error>().to_string().as_string().unwrap()));
+        };
+        #[cfg(feature="std_web")]
+        if let Err(ws_error) = &ws {
+            return Err(match ws_error {
+                CreationError::SyntaxError => WebSocketError::CreationError("Syntax error".to_string()),
+                CreationError::SecurityError => WebSocketError::CreationError("Security error".to_string()),
+            });
+        };
 
-        let ws = ws.map_err(|_| "failed to build websocket")?;
+        let ws = ws.unwrap();
```

Comment:
Do you mind rewriting without the `unwrap()`? You could do something like `let ws = ws.map_err(...)?;`

Reply:
Will do.